next/src/field/type.ts

```diff
@@ -1,47 +1,164 @@
 import type { JsfSchemaType } from '../types'
 
-/**
- * WIP type for UI field output that allows for all `x-jsf-presentation` properties to be splatted
- * TODO/QUESTION: what are the required fields for a field? what are the things we want to deprecate, if any?
- */
-export interface Field {
-  name: string
-  label?: string
-  description?: string
-  fields?: Field[]
-  // @deprecated in favor of inputType,
+export type FieldType = 'text' | 'number' | 'select' | 'file' | 'radio' | 'group-array' | 'email' | 'date' | 'checkbox' | 'fieldset' | 'money' | 'country' | 'textarea'
+
+interface BaseField {
   type: FieldType
-  inputType: FieldType
+  name: string
+  label: string
   required: boolean
+  inputType: FieldType
   jsonType: JsfSchemaType
+  errorMessage: Record<string, string>
+  schema: any
   isVisible: boolean
-  accept?: string
-  errorMessage?: Record<string, string>
-  computedAttributes?: Record<string, unknown>
+}
+
+export interface FieldOption {
+  label: string
+  value: string
+  description?: string
+}
+
+export interface FieldSelect extends BaseField {
+  type: 'select'
+  options: FieldOption[]
+}
+
+export interface FieldTextarea extends BaseField {
+  type: 'textarea'
+  maxLength?: number
+  minLength?: number
+}
+
+export interface FieldDate extends BaseField {
+  type: 'date'
+  format: string
   minDate?: string
   maxDate?: string
   maxLength?: number
-  maxFileSize?: number
-  format?: string
-  anyOf?: unknown[]
-  options?: unknown[]
-  const?: unknown
-  checkboxValue?: unknown
+}
 
-  // Allow additional properties from x-jsf-presentation (e.g. meta from oneOf/anyOf)
-  [key: string]: unknown
+export interface FieldText extends BaseField {
+  type: 'text'
+  maxLength?: number
+  maskSecret?: number
+  pattern?: string
 }
 
-/**
- * Field option
- * @description
- * Represents a key/value pair that is used to populate the options for a field.
- * Will be created from the oneOf/anyOf elements in a schema.
- */
-export interface FieldOption {
+export interface FieldRadio extends BaseField {
+  type: 'radio'
+  options: FieldOption[]
+  direction?: 'row' | 'column'
```

Comment:
todo: Here's another example. This is Remote needs, not a JSF concern. A few other examples down the line (currency, fileDownload, etc... All of those types can exist, but not at JSF.

Reply:
You're right 💯 ! Probably the right move is to extend these types on Remote SDK project which uses Remote JSON Schemas 👍